tutorials/tools/Data_Simulator.ipynb

```diff
@@ -0,0 +1,347 @@
+    "\n",
+    "The simulator includes a multi-microphone far-field mode that incorporates synthetic room impulse response generation in order to simulate multi-microphone multispeaker sessions. When using RIR generation, the RIR is computed once per batch of sessions, and then each constructed utterance is convolved with the RIR in order to get the sound recorded by each microphone before adding the utterance to the audio session. In this tutorial, only near field sessions will be generated.\n",
+    "\n",
+    "The simulator also has a speaker enforcement mode which ensures that the correct number of speakers appear in each session, since it is possible that fewer than the desired number may be present since speaker turns are probabilistic. In speaker enforcement mode, the length of the session or speaker probabilites may be adjusted to ensure all speakers are included before the session finishes."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Parameters"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from omegaconf import OmegaConf\n",
+    "conf = OmegaConf.load('NeMo/tools/speech_data_simulator/conf/data_simulator.yaml')\n",
+    "print(OmegaConf.to_yaml(conf))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Step 4: Create Background Noise Manifest"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!find RIRS_NOISES/real_rirs_isotropic_noises/*.wav > bg_noise.list \\\n",
+    "&& path=\"$(pwd)/NeMo\" \\\n",
+    "&& export PYTHONPATH=$path:$PYTHONPATH \\\n",
+    "&& python NeMo/scripts/speaker_tasks/pathfiles_to_diarize_manifest.py \\\n",
+    "--paths2audio_files bg_noise.list  \\\n",
+    "--manifest_filepath bg_noise.json"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Step 5: Generate Simulated Audio Session\n",
+    "\n",
+    "A single 4-speaker session of 60 seconds is generated as an example."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!path=\"$(pwd)/NeMo\" \\\n",
+    "&& export PYTHONPATH=$path:$PYTHONPATH \\\n",
+    "&& python NeMo/tools/speech_data_simulator/multispeaker_simulator.py \\\n",
+    "  --config-path='conf' --config-name='data_simulator.yaml' \\\n",
+    "  data_simulator.random_seed=42 \\\n",
+    "  data_simulator.manifest_filepath=\"dev-clean-align.json\" \\\n",
+    "  data_simulator.outputs.output_dir=\"test_closetalk\" \\\n",
+    "  data_simulator.session_config.num_sessions=1 \\\n",
+    "  data_simulator.session_config.session_length=30  \\\n",
+    "  data_simulator.background_noise.background_manifest=\"bg_noise.json\""
```

Comment:
This part, we have talked about changing this to python function.
To do that, copy these lines (Just plug in your python files):
Also, assume that this code is already merged to main, and test with your chooper1 branch in your git.
```
from omegaconf import OmegaConf
MODEL_CONFIG = os.path.join(data_dir,'offline_diarization.yaml')
if not os.path.exists(MODEL_CONFIG):
    config_url = "https://raw.githubusercontent.com/NVIDIA/NeMo/main/examples/speaker_tasks/diarization/conf/offline_diarization.yaml"
    MODEL_CONFIG = wget.download(config_url,data_dir)

config = OmegaConf.load(MODEL_CONFIG)
print(OmegaConf.to_yaml(config))

pretrained_speaker_model='titanet_large'
config.diarizer.manifest_filepath = 'data/input_manifest.json'
config.diarizer.out_dir = output_dir # Directory to store intermediate files and prediction outputs

config.diarizer.speaker_embeddings.model_path = pretrained_speaker_model
config.diarizer.speaker_embeddings.parameters.window_length_in_sec = 1.5
config.diarizer.speaker_embeddings.parameters.shift_length_in_sec = 0.75
config.diarizer.oracle_vad = True # ----> ORACLE VAD 
config.diarizer.clustering.parameters.oracle_num_speakers = True

from nemo.collections.asr.models import ClusteringDiarizer
oracle_model = ClusteringDiarizer(cfg=config)
oracle_model.diarize()

```
The original code is from speaker diarization tutorial notebook:
https://github.com/NVIDIA/NeMo/blob/main/tutorials/speaker_tasks/Speaker_Diarization_Inference.ipynb